src/client.js

```diff
@@ -60,9 +70,14 @@ import './styles/main.pcss'
         renderApp(state, emitter)
       })
     })
-
+    
     renderApp(window.__APP_INITIAL_STATE__, emitter)
-
+    
+    const hash = window.location.hash.substr(1)
```

Comment:
Again we should probably check if hash is not empty to ensure we are not using an empty query selector.